### PR TITLE
Final group capstone - Yacht Reservation App

### PR DESCRIPTION
**Pull request by akos29**
Final group capstone - Yacht Reservation App
--
Hello 👋,

In this PR, we excited to introduce our yacht booking app. We've put in a lot of effort to bring this project to life. Here are the key highlights of what we've achieved:

- Repository Setup:

  - [x] Created separate repositories for the front-end and back-end.
  - [x] Organized code structure for scalability and maintainability.
  - [x] Dependencies Installed
  - [x]  Installed all necessary dependencies for both the front-end and back-end, ensuring a smooth development process.
  - [x] Use `git flow` , `linters` and `rubocop`

- API Endpoints:
  - [x] Developed robust API endpoints in the back-end to support yacht booking functionalities.
  - [x]   Defined routes, controllers, and models for seamless data management.
- Front-End Pages:
  - [x]  Crafted user-friendly front-end pages for the yacht booking app.
  - [x] Employed responsive design principles for optimal user experience.
  - [x] Styled pages using Bootstrap and custom CSS for an appealing look.
- API Integration
  - [x]  Established seamless communication between the front-end and back-end through API requests.
  - [x] - Dynamically fetched and displayed content based on API responses.
- Testing:
  - [x]  tested both the front-end and back-end to ensure the reliability and functionality of our app.
  - [x] Utilized appropriate testing frameworks for efficient testing workflows.
API Documentation:
- [x] Created comprehensive API documentation using Swagger, making it easier for developers to understand and use our APIs.

Link to the Back-End's [PR:](https://github.com/akos29/book_appointment/pull/32)

We look forward to your review and feedback on our yacht booking app. Thank you for your time and support! 😊

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,44 +1,10 @@
...
-For more information on the importance of a professional README for your repositories: https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/readme_best_practices.md
--->
-
 <div align="center">
-  <!-- You are encouraged to replace this logo with your own! Otherwise you can also remove it. -->
-  <img src="murple_logo.png" alt="logo" width="140"  height="auto" />
-  <br/>
 
-  <h3><b>Microverse README Template</b></h3>
+  <h3><b>YoYachts</b></h3>
```
Comment (Meltrust) [inline README.md:12]: - [x] This is the default test that comes with React, kindly remove it:

image

File: src/App.js [hunk 2]
```diff
@@ -1,10 +1,79 @@
...
+import Home from './pages/Index';
+import Login from './pages/Login';
+import Register from './pages/Register';
+import AddYacht from './components/AddYacht';
+import Nav from './components/Nav';
+import DeleteYacht from './components/DeleteYacht';
+import 'react-toastify/dist/ReactToastify.css';
+import Reserve from './components/reservation/Index';
 
 function App() {
```
Comment (Meltrust) [inline src/App.js]: - [x]  Please, kindly refactor this main App component to ES6 arrow function, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

That way, your code will meet the latest standards. 💪

Comment (Meltrust) [inline src/App.js]: - [x] Since I wasn't able to run the backend, I can't test the frontend functionality.  This will be taken a look in the next review.

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @akos29,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!










#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: src/components/reservation/ReservationForm.jsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+                required
+                className='w-full px-3 py-2 border rounded-lg focus:outline-none focus:border-blue-500'
+              >
+                <option value=''>Select Yacht Name</option>
+                {yachts.map((yacht) => (
+                  <option key={yacht.id} value={yacht.id}>
+                    {yacht.model}
+                  </option>
+                ))}
+              </select>
```
Comment (mcrd25) [inline src/components/reservation/ReservationForm.jsx:138]: person should really know what they are reserving, image should be visible so they can be sure it's the right yacht they are reserve. [OPTIONAL]

File: src/components/DeleteYacht.jsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      )}
+
+      <DeleteYachtConfirmation
+        isOpen={isConfirmationOpen}
+        onCancel={closeConfirmation}
+        onConfirm={handleConfirmDelete}
+      />
+    </div>
+  );
+};
```
Comment (mcrd25) [inline src/components/DeleteYacht.jsx:109]: Consider limiting 'DELETE YACHT' functionality to yachts the person created themselves or admin user.

File: src/components/AddYacht.jsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        <div className='mb-4'>
+          <label
+            htmlFor='price'
+            className='block text-gray-700 font-semibold mb-2'
+          >
+            Price
+            <input
+              type='number'
+              placeholder='Price'
+              value={price}
```
Comment (mcrd25) [inline src/components/AddYacht.jsx:112]: Should I really be able to set price at 0? Maybe you cna limit this on back-end or front-end side [OPTIONAL]

File: src/components/reservation/ReservationForm.jsx [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+            City
+            <input
+              type='text'
+              id='city'
+              value={city}
+              onChange={(e) => setCity(e.target.value)}
+              required
+              className='w-full px-3 py-2 border rounded-lg focus:outline-none focus:border-blue-500'
+            />{' '}
+          </label>
```
Comment (mcrd25) [inline src/components/reservation/ReservationForm.jsx]: [OPTIONAL] Consider valdiating city chosen. I should not be able to put xyz. Maybe there is an API yo ucan use to validate cities or another way?

**Review by mcrd25 — CHANGES_REQUESTED** [review]
Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Thanks for making changes requested by the previous reviewer

 ## Required Changes ♻️
_Check the comments under the review._
- Don't forget that the application needs to look exactly like design with exception of photos, titles and descriptions. A prime example is the yachts page. I see no carousel functionality like the original design, this is a big deviation. While I love the look of your app, what happens when even more yacths are added?

## Optional suggestions
- The reserve modal close option is not immediately visible to user, consider making it more visible with font thickness and size(maybe use X instead) or adding the close icon on the form modal itself. 
image
*  I am not seeing tests for your Front-End code - this is something we expect to be implemented for the final evaluation. Here are some resources you can use:
   *  https://reactjs.org/docs/testing.html
   * https://jestjs.io/docs/tutorial-react
   * https://www.freecodecamp.org/news/testing-react-hooks/

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Hope all is well, and that my comments were helpful. 😸**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
_Also, you do not have to implement all requested changes, as this is up to you in the end, but I strongly recommend you implement them._

**Pull request merged**